pytorch_lightning/trainer/trainer.py

```diff
@@ -92,6 +92,9 @@ class Trainer(
     DeprecatedTrainerAttributes,
 ):
 
+    accelerator_connector_cls = AcceleratorConnector
+    checkpoint_connector_cls = CheckpointConnector
```

Comment:
So it looks like in this "other library" you will set a different class on the trainer right, one that you subclassed?
In the future, I can see that you will do more of these changes, like other connectors, loops, and so on. Is that the plan?

Reply:
I'm very interested in this too since our guidance so far has been to not extend the trainer class.

Reply:
There's an issue, with this design, the arguments for `AcceleratorConnector` and `CheckpointConnector` of PyTorch Lightning would need to match the other library's, which could cause issues down the road.

Reply:
I don't think making connectors part of the public API makes sense to user. Users should only need to pass the constructed accelerator or checkpoint callback objects.

IMO supporting this other library should not sacrifice or make the existing APIs more complex, especially if it's  experimental.

Reply:
> I don't think making connectors part of the public API makes sense to user. Users should only need to pass the constructed accelerator or checkpoint callback objects.

Agreed.

Reply:
Great points ! Removed it entirely.